x-pack/dev-tools/simulate_autoscaling.py

```diff
@@ -20,29 +20,35 @@ def __init__(self, smoothing_factor, autodetect_dynamics_change):
     self.variance = None
     self.smoothing_factor = smoothing_factor
     self.autodetect_dynamics_change = autodetect_dynamics_change
+    self.dynamics_changed_last_time = False
 
   def dynamics_change_detected(self, value, variance):
     """Returns whether if the (value, variance) is more than 10 stddev unlikely."""
     return (self.variance is not None and self.autodetect_dynamics_change
             and abs(value - self.value) ** 2.0 / (self.variance + variance) > 100.0)
 
-  def add(self, value, variance, dynamics_changed=False) -> None:
+  def add(self, value, variance, dynamics_changed_external=False) -> None:
+    if self.variance is None:
+      self.value = value
+      self.variance = variance
+      self.dynamics_changed_last_time = True
+      return
+
     process_variance = variance / self.smoothing_factor
 
-    if dynamics_changed or self.dynamics_change_detected(value, variance):
+    dynamics_changed = dynamics_changed_external or self.dynamics_change_detected(value, variance)
+    if dynamics_changed or self.dynamics_changed_last_time:
       # If we know we likely had a change in the quantity we're estimating or the prediction is 10 stddev off,
       # we inject extra noise in the dynamics for this step.
-      process_variance *= self.smoothing_factor
-      print(f'dynamic changed: value={value} state={self.value} +-/ {self.error()} '
-            f'(external event={dynamics_changed})')
-
-    if self.variance is None:
-      self.value = value
-      self.variance = variance
-    else:
-      gain = (self.variance + process_variance) / (self.variance + process_variance + variance)
-      self.value += gain * (value - self.value)
-      self.variance = (1 - gain) * (self.variance + process_variance)
+      process_variance = value ** 2
```

Comment:
When the dynamics change, I think the previous
```
process_variance = variance
```
makes no sense. Upon a dynamics change, the estimator should be able to jump to the new value, hence the process variance should be the new value squared.